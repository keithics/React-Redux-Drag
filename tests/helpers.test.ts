import {
  getCoordinates,
  isOverlapping,
  localX2ModuleX,
  moduleW2LocalWidth,
  moduleX2LocalX,
  moduleYGutter,
  setMinMax,
  snapToGutter,
} from '../src/lib/helpers';

describe('helpers', () => {
  test('moduleW2LocalWidth', () => {
    const w = 2;
    expect(moduleW2LocalWidth(w)).toMatchSnapshot();
  });

  test('moduleX2LocalX', () => {
    const x = 2;
    expect(moduleX2LocalX(x)).toMatchSnapshot();
  });

  test('localX2ModuleX', () => {
    const x = 2;
    expect(localX2ModuleX(x)).toMatchSnapshot();
  });

  test('moduleYGutter', () => {
    expect(moduleYGutter(1)).toMatchSnapshot();
  });

  test('setMinMax must return the lowest', () => {
    expect(setMinMax(2, 10, 20)).toMatchSnapshot();
  });

  test('setMinMax must return the highest', () => {
    expect(setMinMax(21, 10, 20)).toMatchSnapshot();
  });

  test('setMinMax must return the same number if between min and max', () => {
    expect(setMinMax(15, 10, 20)).toMatchSnapshot();
  });

  test('getCoordinates', () => {
    expect(
      getCoordinates({
        id: 0,
        coord: { x: 0, y: 0, h: 0, w: 0 },
      })
    ).toMatchSnapshot();
  });

  test('isOverlapping must return true if both modules are overlapping', () => {
    const rect1 = { id: 2, coord: { x: 5, y: 180, w: 2, h: 100, realX: 500 } };
    const rect2 = { id: 1, coord: { x: 4, y: 180, w: 2, h: 100, realX: 500 } };
    expect(isOverlapping(rect1, rect2)).toMatchSnapshot();
  });

  test('isOverlapping must return false if both modules are spaced out', () => {
    const rect1 = { id: 2, coord: { x: 5, y: 180, w: 2, h: 100, realX: 500 } };
    const rect2 = { id: 1, coord: { x: 8, y: 180, w: 2, h: 100, realX: 500 } };
    expect(isOverlapping(rect1, rect2)).toMatchSnapshot();
  });

  test('snapToGutter', () => {
    expect(snapToGutter(2)).toMatchSnapshot();
  });
});
