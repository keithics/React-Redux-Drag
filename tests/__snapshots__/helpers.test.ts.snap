// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers getCoordinates 1`] = `
Object {
  "bottom": 0,
  "left": 10,
  "right": 0,
  "top": 0,
}
`;

exports[`helpers isOverlapping must return false if both modules are spaced out 1`] = `false`;

exports[`helpers isOverlapping must return true if both modules are overlapping 1`] = `true`;

exports[`helpers localX2ModuleX 1`] = `-1`;

exports[`helpers moduleW2LocalWidth 1`] = `159`;

exports[`helpers moduleX2LocalX 1`] = `179`;

exports[`helpers moduleYGutter 1`] = `11`;

exports[`helpers setMinMax must return the highest 1`] = `20`;

exports[`helpers setMinMax must return the lowest 1`] = `10`;

exports[`helpers setMinMax must return the same number if between min and max 1`] = `15`;

exports[`helpers snapToGutter 1`] = `10`;
